### ISSUE TITLE
Problem ECommerce with Marten AddProduct

### ISSUE THREAD

**Issue by FelipeBerrios** [oskardudycz/EventSourcing.NetCore#88]
Hello,

I am testing the ECommerce example with Marten, specifically the actions of adding carts and products. When I enter the cart and the first product everything works correctly. The problem appears when adding a second product, as the following error is displayed.

![image](https://user-images.githubusercontent.com/7199448/143939154-f52d26d5-b4e2-427b-b056-7a48ad4750c2.png)

![image](https://user-images.githubusercontent.com/7199448/143939266-c123d39a-9244-4bcf-b433-79761afba72e.png)

It seems to be a problem when trying to aggregate the event stream.

![image](https://user-images.githubusercontent.com/7199448/143939445-4691e405-4329-4275-82a9-21054c2f62b0.png)

I hope you can help me, thank you very much

**Comment by oskardudycz** [comment]
Indeed there is an issue. It seems that I might have broken something during the .NET 6 upgrade. I'll fix that and keep you posted.

Thank you for dropping me a note.

**Comment by oskardudycz** [comment]
@FelipeBerrios, thank you again for spotting that, and I apologise for the confusion. 

It seems that during the .NET upgrade, I broke Newtonsoft Json.NEserialisationon. I removed the usage of `JsonConstructor` and NuGet that extended Json.NET with private constructors. Unfortunately, the test suite was not strong enough. That is strengthened up now.

I added `ContractResolver`, which enables private non-default constructors.

See the details in: https://github.com/oskardudycz/EventSourcing.NetCore/pull/89.

Code on the main branch should be working fine now 👍
